src/PacketaWidget/View.jsx

```diff
@@ -90,58 +95,110 @@ export const View = ( { cart } ) => {
 	}, [ dynamicSettings, setDynamicSettings, viewState, setViewState, shippingAddress ] );
 
 	const onWidgetButtonClicked = useOnWidgetButtonClicked(
-		packetaShippingRate,
+		packetaPickupPointShippingRate,
 		settings,
 		dynamicSettings,
 		setViewState,
 		shippingAddress,
 		cartItemsWeight,
 	);
 
+	const onHDWidgetButtonClicked = useOnHDWidgetButtonClicked(
+		packetaHomeDeliveryShippingRate,
+		settings,
+		dynamicSettings,
+		setViewState,
+		shippingAddress,
+	);
+
 	useEffect( () => {
 		if (
-			packetaShippingRate &&
+			packetaPickupPointShippingRate &&
 			dynamicSettings &&
 			! viewState &&
 			widgetAutoOpen
 		) {
 			onWidgetButtonClicked();
 		}
-	}, [ packetaShippingRate, widgetAutoOpen, onWidgetButtonClicked ] );
+	}, [ packetaPickupPointShippingRate, widgetAutoOpen, onWidgetButtonClicked ] );
 
-	const getErrorMessage = function ( viewState ) {
+	const getPickupPointErrorMessage = function ( viewState ) {
 		if ( viewState && viewState.pickupPoint ) {
 			return null;
 		} else {
 			return translations.pickupPointNotChosen;
 		}
 	};
 
-	const { choosePickupPoint, packeta } = translations;
+	const getHomeDeliveryErrorMessage = function ( viewState, addressValidationSetting ) {
+		if ( addressValidationSetting === 'optional' || ( viewState && viewState.deliveryAddressInfo ) ) {
+			return null;
+		} else if ( viewState && viewState.deliveryAddressError ) {
+			return viewState.deliveryAddressError;
+		} else {
+			return translations.addressIsNotValidatedAndRequiredByCarrier;
+		}
+	};
 
-	if ( ! packetaShippingRate ) {
-		return null;
+	const { choosePickupPoint, chooseAddress, packeta } = translations;
+
+	if ( packetaPickupPointShippingRate ) {
```

Comment:
asi bych celou tu logiku nad tím vytáhl mimo a nechal tu jen returny s komponentama ať nám to zbytečně nebobtná